### ISSUE TITLE
Keycloak should not send HSTS headers (Strict-Transport-Security) in dev mode and/or when host is localhost

### ISSUE THREAD

**Issue by DamianEdwards** [keycloak/keycloak#32366]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

core

### Describe the bug

Keycloack currently defaults to sending HSTS headers (`Strict-Transport-Security`) for all realms, including the master realm and newly created realms. This in itself is not an issue, except that it does it even when the host name is `localhost`, which pollutes the HSTS policy for all sites served on the local machine via `localhost`. This is pretty annoying on developer machines as it instructs the browser to always attempt a redirect of any request to `http://localhost:[port]` to `https://localhost:[port]` which generally fails, as the same port can't serve both HTTP and HTTPS.

See #14162 and #32319 for other reports of this.

### Version

25.0.4

### Regression

- [ ] The issue is a regression

### Expected behavior

Keycloak should not send HSTS headers when the host is `localhost`. Additionally, it should consider not sending them at all when running in dev mode, or add a setting that allows this to be controlled.

### Actual behavior

Keycloak sends HSTS headers on localhost by default for all realms, which pollutes the browser HSTS policy store for all sites being served locally on localhost in development environments.

### How to Reproduce?

Configure Keycloak to use HTTPS and HTTP and start it in dev mode. Navigate to the home page on HTTPS and see that the response includes the `Strict-Transport-Security` header. Now try to navigate to the home page on the HTTP endpoint and see that the browser auto-redirects (307) to the HTTP endpoint but with a scheme of `https://`, resulting in an error.

**Comment by vmuzikar** [comment]
@DamianEdwards Thank you for the report.

I see this rather as an enhancement than a bug. Also re-assigning to the core team.

Thinking about it a bit more, I agree we could improve the dev UX it might make sense to selectively exclude the HSTS header for localhost if it's present in the Realm headers.

CC @keycloak/core

**Comment by rmartinc** [comment]
@DamianEdwards You can configure the HSTS header in the Realm Settings -> Security Defenses -> HTTP Strict Transport Security (HSTS). You can delete it to not send anything for testing purposes. Do you think that we also need to make this more complicated? Right now it just send what is configured there.

**Comment by DamianEdwards** [comment]
@rmartinc yes I realize it can be configured, but by default (new realms, all domains) it sends the headers for localhost, which essentially pollutes the entire localhost domain, which is a huge pain for development scenarios. IMO HSTS should simply never be sent on localhost in the default configuration.
